From: Peter Zijlstra <peterz@infradead.org>
Date: Fri, 22 Nov 2024 12:47:48 +0100
Subject: mm/gup: Use raw_seqcount_try_begin()
Git-commit: 7528585290a1a1d4e0fb4b72261eb2d8c85de2d7
Patch-mainline: v6.14-rc1
References: bsc#1236648

David pointed out that gup_fast() does exactly what the new
raw_seqcount_try_begin() does -- use it.

Suggested-by: David Hildenbrand <david@redhat.com>
Signed-off-by: Peter Zijlstra (Intel) <peterz@infradead.org>
Reviewed-by: David Hildenbrand <david@redhat.com>
Signed-off-by: Vlastimil Babka <vbabka@suse.cz>
---
 mm/gup.c |    3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

--- a/mm/gup.c
+++ b/mm/gup.c
@@ -3360,8 +3360,7 @@ static unsigned long gup_fast(unsigned l
 		return 0;
 
 	if (gup_flags & FOLL_PIN) {
-		seq = raw_read_seqcount(&current->mm->write_protect_seq);
-		if (seq & 1)
+		if (!raw_seqcount_try_begin(&current->mm->write_protect_seq, seq))
 			return 0;
 	}
 
